gadgets/trace_exec/test/trace_exec_test.go

```diff
@@ -79,8 +81,10 @@ func TestTraceExec(t *testing.T) {
 	cmd := fmt.Sprintf("cp /bin/sh /usr/bin/sh ; setuidgid 1000:1111 /usr/bin/sh -c '%s'", innerCmd)
 	shArgs := []string{"/bin/sh", "-c", cmd}
 	innerShArgs := []string{"/usr/bin/sh", "-c", innerCmd}
+	cwdTestCmd := fmt.Sprintf("cd /tmp ; %s", strings.Join(sleepArgs, " "))
 
 	testContainer := containerFactory.NewContainer(containerName, cmd, containerOpts...)
+	testCwdContainer := containerFactory.NewContainer(cwdTestContainerName, cwdTestCmd, containerOpts...)
```

Comment:
The tests fail in kubernetes because it creates 2 different pods and for each of them it creates a Kubernetes namespace with the same name. See `containers.WithContainerNamespace` a few lines above.

So currently the sequence is as follows:
- namespace/test-trace-exec-402149015484310388 created
- pod/test-trace-exec condition met
- namespace/test-trace-exec-402149015484310388 unchanged
- pod/test-trace-exec-cwd condition met
- pod "test-trace-exec-cwd" deleted
- namespace "test-trace-exec-402149015484310388" deleted
- Error from server (NotFound): pods "test-trace-exec" not found

When deleting the Kubernetes namespace for the first time, it deletes the pod that remained inside. Then, at the next step, it fails to delete the pod.

This can be fixed by using two different Kubernetes namespaces for each pod.

Reply:
Wasn't sure how it would work with another namespace for kubectl-gadget. So changed the already exist container to test cwd too